File: ListView Demo/MainActivity.java
package com.example.demo11;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final ListView listView1 = (ListView) findViewById(R.id.listView1);

        //String Array of values

        String [] data = new String[]{
                "C", "C++", "Java", "VB.NET", "C#", "Node.js", "JavaScript", "HTML",
                "CSS", "PHP"
        };

        ArrayAdapter<String> adapter = new ArrayAdapter<String>
                (this, android.R.layout.simple_list_item_1, data);

        listView1.setAdapter(adapter);


        listView1.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {

                String currentdata = listView1.getItemAtPosition(i).toString();

                Toast.makeText(MainActivity.this, currentdata, Toast.LENGTH_SHORT).show();
                Toast.makeText(MainActivity.this, String.valueOf(l), Toast.LENGTH_SHORT).show();

            }
        });

    }
}

File: ListView Demo/MainActivity.kt
package com.example.demo11

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val listView1 = findViewById<ListView>(R.id.listView1)
        val data = arrayOf(
            "C", "C++", "Java", "VB.NET", "C#", "Node.js", "JavaScript", "HTML",
            "CSS", "PHP"
        )
        val adapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, data)
        listView1.adapter = adapter
        listView1.setOnItemClickListener { _, _, i, l ->
            val currentData = listView1.getItemAtPosition(i).toString()
            Toast.makeText(this, currentData, Toast.LENGTH_SHORT).show()
            Toast.makeText(this, l.toString(), Toast.LENGTH_SHORT).show()
        }
    }
}
